### Lab | Hypothesis Testing

1. It is assumed that the mean systolic blood pressure is **μ = 120 mm Hg**.
In the Honolulu Heart Study, a sample of **n = 100 people** had an average systolic blood pressure of **130.1 mm Hg** with a **standard deviation of 21.21 mm Hg**.
Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

Set up the hypothesis test

H0(null): µ=120 

Ha(alt):  µ != 120

one-sided test: Ha:µ ≠ 120, H0 = False

data should be normally distributed, 
following a t-distribution with
        n-1 degrees of freedom: DF = 100-1= 99
        sample error: SE=130.1-120=10.1
        confidence level: 95%, significance level: α=0.05

µsam=130.1, µ0=120, S(sigma)=21.21, 

In [43]:
from scipy.stats import ttest_1samp
import scipy.stats as stats
import math
import numpy as np
import pandas as pd
import statistics as std


In [19]:
t = (130.1-120)/(21.21/math.sqrt(100))
t=round(t, 4)
print("test statistics t is", t)

test statistics t is 4.7619


critic value from the Student's t distribution table with alpha=0.05 and (n-1) dregrees of freedom:
t(0.025,99)=1.9842

In [23]:
tc = stats.t.ppf(.975, df=99)
tc =round(tc, 5)
print("The critical value for df=99 and confidence lvl of 97.5% is", tc)


The critical value for df=99 and confidence lvl of 97.5% is 1.98422


Compare both values and make decision: 4.7619> 1.98422, Nullhypotheses **rejected**

In [27]:
x = np.random.normal(loc=130.1, scale=21.21, size=100)

In [25]:
tscore, pvalue = ttest_1samp(x, popmean=120)
print("t Statistic: ", tscore)  
print("P Value: ", pvalue)

t Statistic:  3.305712247282784
P Value:  0.0013203428083353981


2. In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file Data/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other

In [28]:
#dataframe for the time (in sec) each machine needs to pack a carton of 10 jars
d = {'new_machine': [42.1, 41, 41.3, 41.8, 42.4, 42.8, 43.2, 42.3, 41.8, 42.7], 'old_machine': [42.7, 43.6, 43.8, 43.3, 42.5, 43.5, 43.1, 41.7, 44, 44.1]}
df = pd.DataFrame(data=d)

H0: μ1 ≠ μ2

Ha: μ1 = μ2

Note: The null hypothesis says that the new machine is better. If this is the case, then the mean of the two samples would not be equal. That is the null hypothesis

In [58]:
#calculate mean of the two groups
m1=np.mean(df.new_machine)
m2=np.mean(df.old_machine)
print('mean of the old machine ', m1)
print('mean of the new machine ', round(m2,2))

mean of the old machine  42.14
mean of the new machine  43.23


In [70]:
#Calculate the variance for each group
s1=np.var(df.new_machine, ddof=1)
s2=np.var(df.old_machine, ddof=1)
print('s1 of the old machine ', round(s1,5))
print('s2 of the new machine ', round(s2,5))

s1 of the old machine  0.46711
s2 of the new machine  0.56233


In [71]:
t=(m1-m2)/(math.sqrt(((s1)/10)+((s2)/10)))
t

-3.397230706117603

In [72]:
stats.ttest_ind(df.new_machine, df.old_machine)

Ttest_indResult(statistic=-3.3972307061176026, pvalue=0.0032111425007745158)

Conclusion: Significant difference in the two means, the new machine is signicantly faster than the old one.